providers/src/airflow/providers/http/hooks/http.py

```diff
@@ -83,11 +87,25 @@ def __init__(
         self.base_url: str = ""
         self._retry_obj: Callable[..., Any]
         self._auth_type: Any = auth_type
+
+        if adapter is not None and not isinstance(adapter, BaseAdapter):
+            raise TypeError("adapter must be an instance of requests.adapters.BaseAdapter")
+        self.adapter = adapter
+
         self.tcp_keep_alive = tcp_keep_alive
         self.keep_alive_idle = tcp_keep_alive_idle
         self.keep_alive_count = tcp_keep_alive_count
         self.keep_alive_interval = tcp_keep_alive_interval
 
+        if self.tcp_keep_alive:
+            self.keep_alive_adapter = TCPKeepAliveAdapter(
+                idle=self.keep_alive_idle,
+                count=self.keep_alive_count,
+                interval=self.keep_alive_interval,
+            )
+        else:
+            self.keep_alive_adapter = None
```

Comment:
What's this all about?

Reply:
Ah just moved. 

This doesn't make any network requests by creating it here does it?

Reply:
it doesn't, adapters just modify how actual requests are handled